core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -803,17 +804,18 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
       // Possible ufs sync.
       if (syncMetadata(rpcContext, inodePath, lockingScheme, DescendantType.ONE)) {
         // If synced, do not load metadata.
-        options.setLoadMetadataType(LoadMetadataType.Never);
+        context.getOptions().setLoadMetadataType(LoadMetadataPType.NEVER);
       }
 
       // If the file already exists, then metadata does not need to be loaded,
       // otherwise load metadata.
       if (!inodePath.fullPathExists()) {
-        checkLoadMetadataOptions(options.getLoadMetadataType(), inodePath.getUri());
+        checkLoadMetadataOptions(context.getOptions().getLoadMetadataType(), inodePath.getUri());
         loadMetadataIfNotExist(rpcContext, inodePath,
-            MASTER_OPTIONS.getLoadMetadataOptions().setCreateAncestors(true).setCommonOptions(
-                MASTER_OPTIONS.getCommonOptions().setTtl(options.getCommonOptions().getTtl())
-                    .setTtlAction(options.getCommonOptions().getTtlAction())));
+            LoadMetadataContext.defaults(LoadMetadataPOptions.newBuilder().setCreateAncestors(true)
+                .setCommonOptions(FileSystemMasterCommonPOptions.newBuilder()
+                    .setTtl(context.getOptions().getCommonOptions().getTtl())
+                    .setTtlAction(context.getOptions().getCommonOptions().getTtlAction()))));
```

Comment:
probably worth saving `context.getOptions()` and `context.getOptions.getCommonOptions()` to local variables since they are used many times.